tests/integration/metrics/multicolumn_values/test_sum_equal.py

```diff
@@ -0,0 +1,147 @@
+        assert result.value == SUCCESS_COUNT
+
+    @parameterize_batch_for_data_sources(
+        data_source_configs=PANDAS_DATA_SOURCES + SPARK_DATA_SOURCES + SQL_DATA_SOURCES,
+        data=DATA_FRAME,
+    )
+    def test_failure(self, batch_for_datasource: Batch) -> None:
+        batch = batch_for_datasource
+        metric = MultiColumnSumEqualUnexpectedCount(
+            sum_total=FAILURE_SUM_TOTAL,
+            column_list=[COL_A, COL_B],
+        )
+        result = batch.compute_metrics(metric)
+        assert isinstance(result, MultiColumnSumEqualUnexpectedCountResult)
+        assert result.value == FAILURE_COUNT
+
+    @pytest.mark.parametrize(
+        "ignore_row_if,unexpected_count",
+        [
+            ("any_value_is_missing", 1),
+            ("all_values_are_missing", 3),
+            ("never", 6),
+        ],
+    )
+    @parameterize_batch_for_data_sources(
+        data_source_configs=PANDAS_DATA_SOURCES,
+        data=DATA_FRAME_WITH_NULLS,
+    )
+    def test_ignore_row_if__pandas(
+        self, batch_for_datasource: Batch, ignore_row_if, unexpected_count
+    ) -> None:
+        batch = batch_for_datasource
+        metric = MultiColumnSumEqualUnexpectedCount(
+            sum_total=FAILURE_SUM_TOTAL,
+            column_list=[COL_A_WITH_NULLS, COL_B_WITH_NULLS],
+            ignore_row_if=ignore_row_if,
+        )
+        result = batch.compute_metrics(metric)
+        assert result.value == unexpected_count
+
+    @pytest.mark.parametrize(
+        "ignore_row_if,unexpected_count",
+        [
+            pytest.param(
+                "any_value_is_missing", 1, marks=pytest.mark.xfail(reason="returns 6", strict=True)
+            ),
+            pytest.param(
+                "all_values_are_missing",
+                3,
+                marks=pytest.mark.xfail(reason="returns 6", strict=True),
+            ),
+            pytest.param("never", 6),
+        ],
+    )
+    @parameterize_batch_for_data_sources(
+        data_source_configs=SPARK_DATA_SOURCES,
+        data=DATA_FRAME_WITH_NULLS,
+    )
+    def test_ignore_row_if__spark(
+        self, batch_for_datasource: Batch, ignore_row_if, unexpected_count
+    ) -> None:
+        batch = batch_for_datasource
+        metric = MultiColumnSumEqualUnexpectedCount(
+            sum_total=FAILURE_SUM_TOTAL,
+            column_list=[COL_A_WITH_NULLS, COL_B_WITH_NULLS],
+            ignore_row_if=ignore_row_if,
+        )
+        result = batch.compute_metrics(metric)
+        assert result.value == unexpected_count
+
+    @pytest.mark.parametrize(
+        "ignore_row_if,unexpected_count",
+        [
+            pytest.param("any_value_is_missing", 1),
+            pytest.param(
+                "all_values_are_missing",
+                3,
+                marks=pytest.mark.xfail(reason="returns 1", strict=True),
+            ),
+            pytest.param("never", 6, marks=pytest.mark.xfail(reason="returns 1", strict=True)),
```

Comment:
Or maybe since this fails on SQL this is a real bug

Reply:
right, hard to say - i've pushed the explicit types to see if that helps

Reply:
we can't use explicit types as pytest parameters, because they belong to the Spark compatibility layer and need to exist in the top level namespace, and so blow up in the unit tests.

Reply:
I had to do [this](https://github.com/great-expectations/great_expectations/blob/325133d85920cb7bd271d4c65422a8069e560140/tests/integration/data_sources_and_expectations/test_expectation_conditions.py#L26) in the past